### PR TITLE
ceph-volume: add --version

### PR DESCRIPTION
**Pull request by sebastian-philipp**
Signed-off-by: Sebastian Wagner <sewagner@redhat.com>

Fixes: https://tracker.ceph.com/issues/38044

The idea is to use the same information for `cephadm --version`

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/ceph-volume/ceph_volume/main.py [hunk 1]
```diff
@@ -137,6 +142,11 @@ def main(self, argv):
         log.setup_console()
         logger = logging.getLogger(__name__)
         logger.info("Running command: ceph-volume %s %s", " ".join(main_args), " ".join(subcommand_args))
+
+        if args.version:
+            print('ceph version {0}'.format(ceph_volume.__version__))
+            return 0
```
Comment (tchaikov) [inline src/ceph-volume/ceph_volume/main.py:148]: the entry point of the ceph-volume cli is `Volume.__init__()`, which return `None`. i think the same applies to `Volume.main()`.  probably we should just throw `SystemExit(0)` in this case ?

Comment (sebastian-philipp) [reply src/ceph-volume/ceph_volume/main.py]: returning `None` here in this case should be enough

**Review by p-se — APPROVED** [review]
lgtm

File: src/python-common/ceph/version.py.in [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# replaced by cmake:
```
Comment (tchaikov) [inline src/python-common/ceph/version.py.in:1]:
```suggestion
# generated with cmake
```

but i'd suggest remove this comment.

File: src/python-common/CMakeLists.txt [hunk 3]
```diff
@@ -1,3 +1,6 @@
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/ceph/version.py.in
+  ${CMAKE_CURRENT_SOURCE_DIR}/ceph/version.py @ONLY)
```
Comment (tchaikov) [inline src/python-common/CMakeLists.txt:2]: please do not create source file in `src`. put them in `${CMAKE_CURRENT_BINARY_DIR}` instead.

Comment (sebastian-philipp) [reply src/python-common/CMakeLists.txt:2]: This requirement is killing this whole approach unfortunately. Let me think of an alternative approach.

Comment (p-se) [reply src/python-common/CMakeLists.txt:2]: @tchaikov do you think the same applies to https://github.com/ceph/ceph/pull/41855?

**Pull request closed without merging**